readthedocs_build/config/config.py

```diff
@@ -408,6 +413,62 @@ def validate_formats(self):
 
         return True
 
+    def validate_submodules(self):
+        raw_submodules = self.raw_config.get('submodules')
+        if raw_submodules is None:
+            self['submodules'] = self.get_default_submodules_config()
+            return None
+
+        if 'include' in raw_submodules and 'exclude' in raw_submodules:
+            self.error(
+                'submodules',
+                'It is not possible to include and exclude submodules',
+                code=INVALID_KEYS_COMBINATION
+            )
+
+        self['submodules'] = {}
+        submodules = self['submodules']
+
+        with self.catch_validation_error('submodules.recursive'):
+            recursive = raw_submodules.get('recursive', False)
+            submodules['recursive'] = validate_bool(recursive)
+
+        if 'include' in raw_submodules:
+            with self.catch_validation_error('submodules.include'):
+                include = raw_submodules['include']
+                if include == ALL:
+                    submodules['include'] = ALL
+                else:
+                    submodules['include'] = validate_list(include)
+                    self.validate_directories(submodules['include'])
+                if not submodules['include']:
+                    submodules['recursive'] = False
+        elif 'exclude' in raw_submodules:
+            with self.catch_validation_error('submodules.exclude'):
+                exclude = raw_submodules['exclude']
+                submodules['exclude'] = validate_list(exclude)
+                self.validate_directories(submodules['exclude'])
+        elif submodules['recursive'] is True:
+            self.error(
+                'submodules',
+                'You need to include submodules',
+                code=MISSING_REQUIRED_KEY
+            )
+        return True
+
+    def get_default_submodules_config(self):
```

Comment:
This would be different for the previous projects

```yaml
submodules:
  include: all
  recursive: true
```

And probably this would be managed by the rtd code...